strawberryfields/api/connection.py

```diff
@@ -89,14 +89,17 @@ def __init__(
     ):
         default_config = load_config()
 
+        print(token)
         self._token = token or default_config["api"]["authentication_token"]
         self._host = host or default_config["api"]["hostname"]
         self._port = port or default_config["api"]["port"]
         self._use_ssl = use_ssl or default_config["api"]["use_ssl"]
         self._verbose = verbose
 
         self._base_url = "http{}://{}:{}".format("s" if self.use_ssl else "", self.host, self.port)
-        self._headers = {"Authorization": self.token, "Accept-Version": self.api_version}
+
+        self._headers = {"Accept-Version": self.api_version}
+        self._refresh_access_token()
```

Comment:
It's worth noting, that due to this call from now on instantiating a `Connection` object will already try to establish connection with the server.

As users are encouraged to use the higher level `RemoteEngine`, this would mean that they could face the following:
```python
In [12]: os.environ["SF_API_AUTHENTICATION_TOKEN"] = "WrongToken"

In [13]: sf.RemoteEngine("X8")
---------------------------------------------------------------------------
RequestFailedError                        Traceback (most recent call last)
<ipython-input-13-a1e528990489> in <module>
----> 1 sf.RemoteEngine("X8")

~//strawberryfields/strawberryfields/engine.py in __init__(self, target, connection, backend_options)
    548         self._target = self.DEFAULT_TARGETS.get(target, target)
    549         self._spec = None
--> 550         self._connection = connection or Connection()
    551         self._backend_options = backend_options or {}
    552         self.log = create_logger(__name__)

~//strawberryfields/strawberryfields/api/connection.py in __init__(self, token, host, port, use_ssl, verbose)
     99
    100         self._headers = {"Accept-Version": self.api_version}
--> 101         self._refresh_access_token()
    102
    103         self.log = create_logger(__name__)

~//strawberryfields/strawberryfields/api/connection.py in _refresh_access_token(self)
    334             self._headers["Authorization"] = f"Bearer {access_token}"
    335         else:
--> 336             raise RequestFailedError("Authorization failed for request, please check your token provided.")
    337
    338     def _request(self, method: str, path: str, headers: Dict = None, **kwargs):

RequestFailedError: Authorization failed for request, please check your token provided.
```

@josh146 would you think that this causes any complications?

Reply:
Nice. Thanks for adding these @antalszava!